sound/soc/soc-dapm.c

```diff
@@ -4464,6 +4464,14 @@ void snd_soc_dapm_connect_dai_link_widgets(struct snd_soc_card *card)
 			for_each_rtd_codec_dais(rtd, i, codec_dai)
 				dapm_connect_dai_pair(card, rtd, codec_dai,
 						      asoc_rtd_to_cpu(rtd, i));
+		} else if (rtd->dai_link->codec_ch_maps) {
+			int cpu_id;
+
+			for_each_rtd_codec_dais(rtd, i, codec_dai) {
+				cpu_id = rtd->dai_link->codec_ch_maps[i].connected_cpu_id;
+				dapm_connect_dai_pair(card, rtd, codec_dai,
+						      asoc_rtd_to_cpu(rtd, cpu_id));
+			}
```

Comment:
probably want to change the log on line 4477 as "N cpus to M codecs link is not supported: no mapping provided\n"

Reply:
> probably want to change the log on line 4477 as "N cpus to M codecs link is not supported: no mapping provided\n"

Yes, I forgot it. Thanks for the suggestion.